src/scenes/Modals/GenericActionModal.tsx

```diff
@@ -0,0 +1,55 @@
+import { BaseM } from 'components/core/Text/Text';
+import breakpoints from 'components/core/breakpoints';
+import Button from 'components/core/Button/Button';
+import Spacer from 'components/core/Spacer/Spacer';
+import styled from 'styled-components';
+import { useModalActions } from 'contexts/modal/ModalContext';
+import { useCallback } from 'react';
+
+export default function GenericActionModal({
+  action,
+  bodyText,
+  buttonText,
+}: {
```

Comment:
`GenericActionModal` takes in an action, the body text, and the button text. This makes it reusable across all components.

After executing `action`, it runs `hideModal()`

Reply:
This also removes the need for `ConfirmLeaveModal`, which has since been deleted. This modal essentially renavigated to the `/{username}` but, per Linear, we now want to navigate dynamically based on the query parameters.

Now, we handle all exit navigation from the `WizardFooter` and pass in the intended action to `GenericActionModal` (ensuring that the `Back` button and escape key mimic the same behavior.